functions/Remove-DbaServerRoleMember.ps1

```diff
@@ -0,0 +1,194 @@
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Role, Security, Login
+        Author: Mikey Bronowski (@MikeyBronowski), https://bronowski.it
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Remove-DbaServerRoleMember
+
+    .EXAMPLE
+        PS C:\> Remove-DbaServerRoleMember -SqlInstance server1 -ServerRole dbcreator -Login login1
+
+        Removes login1 from the dbcreator fixed server-level role on the instance server1.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaServerRoleMember -SqlInstance server1, sql2016 -ServerRole customrole -Login login1
+
+        Removes login1 from customrole custom server-level role on the instance server1 and sql2016.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaServerRoleMember -SqlInstance server1 -ServerRole customrole -Role dbcreator
+
+        Removes customrole custom server-level role from the dbcreator fixed server-level role.
+
+    .EXAMPLE
+        PS C:\> $servers = Get-Content C:\servers.txt
+        PS C:\> $servers | Remove-DbaServerRoleMember -ServerRole sysadmin -Login login1
+
+        Removes login1 from the sysadmin fixed server-level role in every server in C:\servers.txt.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaServerRoleMember -SqlInstance localhost -ServerRole bulkadmin, dbcreator -Login login1
+
+        Removes login1 from the bulkadmin and dbcreator fixed server-level roles on the server localhost.
+
+    .EXAMPLE
+        PS C:\> $roles = Get-DbaServerRole -SqlInstance localhost -ServerRole bulkadmin, dbcreator
+        PS C:\> $roles | Remove-DbaServerRoleMember -Login login1
+
+        Removes login1 from the bulkadmin and dbcreator fixed server-level roles on the server localhost.
+
+    .EXAMPLE
+        PS C:\ $logins = Get-Content C:\logins.txt
+        PS C:\ $srvLogins = Get-DbaLogin -SqlInstance server1 -Login $logins
+        PS C:\ Remove-DbaServerRoleMember -Login $logins
+
+        Removes all the logins found in C:\logins.txt from mycustomrole custom server-level role on server1.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [parameter(ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [parameter(ValueFromPipeline)]
+        [string[]]$ServerRole,
```

Comment:
Sorry, no time for a test right now. So a question: Does this work? Would you pipe in the name of the server role as string and provide all other info as parameters?

Reply:
It does not work when I pipe in the string

`'customrole' | Remove-DbaServerRoleMember -SqlInstance $s1 -SqlCredential $credential -Login $logins`

it works when piped the object though

```
$r = Get-DbaServerRole -SqlInstance $s1 -SqlCredential $credential -ServerRole $customServerRole
$r | Remove-DbaServerRoleMember -SqlInstance $s1 -SqlCredential $credential -Login $logins
```

If it needs correcting we can make changes to Add-DbaServerRoleMember function as well.

Reply:
But piping in a server role object, this object would not be bound to $ServerRole, but to $InputObject. So I would remove the `[parameter(ValueFromPipeline)]`. And yes, I think it should be changed in Add-DbaServerRoleMember as well.

Reply:
Got it. Removed from the Remove- function.